pkg/k8s/watchers/cilium_envoy_config.go

```diff
@@ -83,6 +84,16 @@ func (k *K8sWatcher) ciliumEnvoyConfigInit(ctx context.Context, ciliumNPClient c
 	k.k8sAPIGroups.AddAPI(k8sAPIGroupCiliumEnvoyConfigV2)
 }
 
+// isIngressKind returns true if any of the OwnerReferences is Kind "Ingress"
+func isIngressKind(meta *meta_v1.ObjectMeta) bool {
+	for _, owner := range meta.OwnerReferences {
+		if owner.Kind == "Ingress" {
+			return true
+		}
+	}
```

Comment:
the same might need to be done for GatewayAPI resources (e.g. Gateway)

Reply:
Let's do that on a separate PR as that would not need to be backported to 1.13, right?

Reply:
@jrajahalme pinging you directly for exposure in #28254. it seems this hasn't been backported for gateway api?